lib/ret/media_resolver.ex

```diff
@@ -64,21 +66,24 @@ defmodule Ret.MediaResolver do
       with client_id when is_binary(client_id) <- resolver_config(:deviantart_client_id),
            client_secret when is_binary(client_secret) <-
              resolver_config(:deviantart_client_secret) do
-        page_resp = retry_get_until_success(uri |> URI.to_string())
+        page_resp = uri |> URI.to_string() |> retry_get_until_success
         deviant_id = Regex.run(@deviant_id_regex, page_resp.body) |> Enum.at(1)
         token_host = "https://www.deviantart.com/oauth2/token"
         api_host = "https://www.deviantart.com/api/v1/oauth2"
 
-        token_url =
+        token =
           "#{token_host}?client_id=#{client_id}&client_secret=#{client_secret}&grant_type=client_credentials"
-
-        token_resp = retry_get_until_success(token_url)
-        token = token_resp.body |> Poison.decode!() |> Kernel.get_in(["access_token"])
-
-        api_url = "#{api_host}/deviation/#{deviant_id}?access_token=#{token}"
-        api_resp = retry_get_until_success(api_url)
-
-        api_resp.body |> Poison.decode!() |> Kernel.get_in(["content", "src"]) |> URI.parse()
+          |> retry_get_until_success
+          |> Map.get(:body)
+          |> Poison.decode!()
+          |> Map.get("access_token")
+
+        "#{api_host}/deviation/#{deviant_id}?access_token=#{token}"
+        |> retry_get_until_success
+        |> Map.get(:body)
+        |> Poison.decode!()
+        |> Kernel.get_in(["content", "src"])
+        |> URI.parse()
```

Comment:
Nice. I can't believe this is how we usually code, assigning things and calling functions like like plebs. Pipe all the things!

Reply:
Haha I know crazy right :)